src/main/java/edu/harvard/iq/dataverse/DataverseLocaleBean.java

```diff
@@ -0,0 +1,144 @@
+package edu.harvard.iq.dataverse;
+
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.Serializable;
+import java.net.URL;
+import java.util.*;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+
+import javax.faces.context.FacesContext;
+import javax.faces.event.ValueChangeEvent;
+import javax.inject.Named;
+import javax.servlet.http.HttpServletRequest;
+
+import org.json.*;
+
+@Named
+@javax.enterprise.context.SessionScoped
+
+public class DataverseLocaleBean implements Serializable {
+
+
+    private static final Logger logger = Logger.getLogger(DataverseLocaleBean.class.getCanonicalName());
+
+    private List<DataverseLocale> dataverseLocales;
+    private static Map<String,Object> languages;
+
+    String json_url= "http://localhost:8080/resources/lang/languages.json";
+
+    JSONObject json_obj;
+
+
+    private String localeCode;
+    private Locale locale;
+
+    FacesContext context = FacesContext.getCurrentInstance();
+
+    {
+        if (context== null){
+            locale= new Locale ("");
+        } else if(context.getViewRoot().getLocale().getLanguage()== "en_US" || context.getViewRoot().getLocale().getLanguage()== "en"){
+            locale= new Locale ("");
+        }else{
+            locale= context.getViewRoot().getLocale();
+        }
+    }
+
+
+    public DataverseLocaleBean(){
+
+    }
+
+    private static String readUrl(String urlString) throws Exception {
```

Comment:
could any tests be added for this?

Reply:
I tested the workflow only with logger messages.